### DISCUSSION TITLE
Compilation: Artist should be used for search instead of Album Artist

### DISCUSSION THREAD

**Discussion by bassstorm** [ghDaYuYu/foo_discogs#109] [General]
**Use case:**
When I call Discogger for a compilation (albumartist = VA) I expect to see search results for its artist(s).

**Expected:**
Search by first artist (or something a bit more smarter) to have Album selection filter actually work.

**Actual:**
Search happens for "VA" with no meaningful results.

P.S. Same for "Various Artists" and "Various".
P.P.S. There's no configuration for that behavior in Searching tab either.

**Comment by ghuDaYuYu** [comment]
As a side note, the album selection filter should work for tracks containing 'Album Title'.
If the value is present and not displayed as the Find Release Filter, then there might be a bug initiating the dialog.

Just to clarify I understand the use case. The files already contain Album Title and track Artists.
And you'd like VA 'Album Artist' being ignored and look up the first track Artist instead, is that right?
If I am getting it right, that´s a pretty good idea!
Can´t tell when a new release will be available... but the feature is now in the todo list.

P.S.

Just wondering about a more elaborated solution...

Ideally, there should be either a check box modifying the search button behavior or a 'Search for releases' one that searches for releases containing multiple artists.

For instance (not a very good example, btw):
https://www.discogs.com/release/203071 features Radanna and Costanza.
Then, filling the Artist/URL search field with Radanna+Costanza and pressing 'Search releases' fetches and adds
'Back to Mine' among a few other releases to the Release Tree panel.

**Comment by bassstorm** [comment]
Yes, you get it right. Normally my files are more or less fully tagged, but lack extra information like label and catalog #. That's often true for compilation albums. I use "VA" as Album Artist for these, but raw files may have "Various Artists" or "Various" as well.

As for your example, yes, I would expect something like that (not sure on exact Discogs REST API capabilities tho). But also maybe the checkbox could be supplied with a little smartness, i.e. configurable _$if()_ switch which should produce that _true/false_ value for the checkbox.

**Comment by ghuDaYuYu** [comment]
Added to TODO list

**Comment by ghuDaYuYu** [comment]
@bassstorm 

There have been some advances related to VAs.

Here is a beta release in case you would like to give it a try.
When the first track is tagged with "VA", or is prefixed with "Various Artists" or "Various" as Album Artist, there is an attempt to get three unique artists from the track selection. The search field is then filled with 'Artist1+Artist2+Artist3' and the 'Search VA' button will try to fetch releases featuring those artists. The role option is disabled for VA searches.

Other fixes:
- Do not throw fatal error loading 404 artist previews.
- Hide brackets in 'Default Master with artist role' and 'Default Release with artist role' (Searching preferences) when roles are not available.

The new feature has been only roughly tested, but beta 3 should prove that your idea looks viable, even though behind the scenes, the search is not bounded by type 'artist' but any content: release, master, artist and label.

Thanks for testing!

[foo_discogger - v1.0.20.beta.3.zip](https://github.com/ghDaYuYu/foo_discogger/files/12480251/foo_discogger.-.v1.0.20.beta.3.zip)

**Comment by ghuDaYuYu** [comment]
This is Beta 4, probably the last beta release before final v1.0.20

Change log:

- Fix VA mode not reverting to normal mode.
- Include experimental feature 'VA release image downloads' (#96 @Lovecraft-github)

Other fixes:

- Preview dialog: crashes on edit completion advancing to an empty row.

From previous betas...

(Beta 3)

- VA used for search instead of Album Artist
- Do not throw fatal error loading 404 artist previews.
- Hide brackets in 'Default Master with artist role' and 'Default Release with artist role' (Searching preferences) when roles are not available.

(Beta 2)

- Close Preview dialog after writing tags

(Beta 1)

- Export tag mappings as JSON

[foo_discogger - v1.0.20.beta.4.zip](https://github.com/ghDaYuYu/foo_discogger/files/12488836/foo_discogger.-.v1.0.20.beta.4.zip)

**Reply by bassstorm** [reply]
Thanks for great news! Tested new build here. Looks good with a minor bump.

These are my use cases for Album Artist value:
- _Various Artists_ - pass (search for first three track artists combined)
- _Various_ - pass (same as above)
- _VA_ - fail (search for VA)

With that I'm wondering if there is any (hidden) configuration to make the last one work as well?

**Comment by ghuDaYuYu** [comment]
Nice, you found it! The 'VA' case was a bug. 
There are no hidden configuration yet, to customize the string pattern.
That will need a configuration version upgrade... probably soon in upcoming releases.

Currently, after this fix, the following patterns should be working:

Album Artist equals to "VA"
Album Artist prefixed "VA "
Album Artist to lowercase prefixed "various"
Album Artist to lowercase prefixed "various artists"
Album Artist to lowercase prefixed "varios"
Album Artist to lowercase prefixed "varios artistas"

[foo_discogger - v1.0.20.beta.5.zip](https://github.com/ghDaYuYu/foo_discogger/files/12489689/foo_discogger.-.v1.0.20.beta.5.zip)

**Reply by bassstorm** [reply]
Confirmed as working. You're the best :)
